spring-integration-core/src/main/java/org/springframework/integration/splitter/AbstractMessageSplitter.java

```diff
@@ -59,10 +68,46 @@ public void setApplySequence(boolean applySequence) {
 		this.applySequence = applySequence;
 	}
 
+	/**
+	 * Specify a channel where rejected Messages should be sent. If the discard
+	 * channel is null (the default), rejected Messages will be dropped.
+	 * A "Rejected Message" means that split function has returned an empty result (but not null):
+	 * no items to iterate for sending.
+	 * @param discardChannel The discard channel.
+	 * @since 5.2
+	 */
+	public void setDiscardChannel(MessageChannel discardChannel) {
+		this.discardChannel = discardChannel;
+	}
+
+	/**
+	 * Specify a channel bean name (resolved to {@link MessageChannel} lazily)
+	 * where rejected Messages should be sent. If the discard
+	 * channel is null (the default), rejected Messages will be dropped.
+	 * A "Rejected Message" means that split function has returned an empty result (but not null):
+	 * no items to iterate for sending.
+	 * @param discardChannelName The discard channel bean name.
+	 * @since 5.2
+	 */
+	public void setDiscardChannelName(String discardChannelName) {
+		Assert.hasText(discardChannelName, "'discardChannelName' must not be empty");
+		this.discardChannelName = discardChannelName;
+	}
+
+	@Override
+	public MessageChannel getDiscardChannel() {
+		String channelName = this.discardChannelName;
```

Comment:
We should exit early if `this.discardChannel` is not null, no need to store this local variable in that case.

Reply:
Also, if both are set the channel should win.